type_inference/ast_traverse.py

```diff
@@ -1,18 +1,67 @@
 import sys
 
 from parser_py import parse
-from type_inference.types import TypesGraph
+from type_inference.types import TypesGraph, UnificationContext
 
 
-def traverse_tree(rule):
+def get_involved_variable_name(expression: dict):
+  while "subscript" in expression:
+    expression = expression["subscript"]["record"]
+
+  return expression["variable"]["var_name"]
+
+
+def connect_unification(types_graph: TypesGraph, left_variable: str,
```

Comment:
Please be consistent Logica's codebase style of CamelCase function names.

Reply:
Yep, it would be changed later